homeassistant/components/debugpy/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""The Remote Python Debugger integration."""
+from asyncio import Event
+import logging
+from threading import Thread
+from typing import Optional
+
+import debugpy
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant, ServiceCall
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.service import async_register_admin_service
+from homeassistant.helpers.typing import ConfigType
+
+DOMAIN = "debugpy"
+CONF_WAIT = "wait"
+CONF_START = "start"
+SERVICE_START = "start"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_HOST, default="0.0.0.0"): cv.string,
+                vol.Optional(CONF_PORT, default=5678): cv.port,
+                vol.Optional(CONF_START, default=True): cv.boolean,
+                vol.Optional(CONF_WAIT, default=False): cv.boolean,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Remote Python Debugger component."""
+    conf = config[DOMAIN]
+
+    async def debug_start(
+        call: Optional[ServiceCall] = None, *, wait: bool = True
+    ) -> None:
+        """Start the debugger."""
+        debugpy.listen((conf[CONF_HOST], conf[CONF_PORT]))
+
+        wait = conf[CONF_WAIT]
+        if wait:
+            _LOGGER.warning(
+                "Waiting for remote debug connection on %s:%s",
+                conf[CONF_HOST],
+                conf[CONF_PORT],
+            )
+            ready = Event()
+
+            def waitfor():
+                debugpy.wait_for_client()
+                hass.loop.call_soon_threadsafe(ready.set)
+
+            Thread(target=waitfor).start()
+
+            await ready.wait()
+        else:
+            _LOGGER.warning(
```

Comment:
Should this be `info`?

Reply:
Yeah, I've considered that. Reason I've opted to keep it at a warning level, because of the following reasons:

- Default logging level. It would show up in the logs. This integration is actively used by developers, I think the feedback is important to see, even when the default log level is used.
- Having the debugger active, is a security risk.

I'm happy to change it to `info` is you think that is a better match?

Reply:
warning makes sense for the reasons above 👍